gsy_framework/sim_results/electric_blue/all_results.py

```diff
@@ -0,0 +1,103 @@
+
+    def update_from_repr(self, area_representation: Dict):
+        """
+        Updates the simulation results using area_representation data that arrive from the d3a-web.
+        """
+
+    @property
+    def all_db_results(self) -> Dict:
+        """Get dict with all the results in format that can be saved to the DB."""
+        results = {"orders": self.orders, "current_device_stats": self.current_device_stats,
+                   "device_time_series": self.device_time_series, "cumulative_net_energy_flow": {},
+                   "cumulative_market_fees": 0.}
+        return results
+
+    @property
+    def total_memory_utilization_kb(self):
+        """Get the total memory allocated by the results."""
+        return self._total_memory_utilization_kb
+
+    def _clear_device_statistics(self) -> None:
+        self.orders.clear()
+        self.current_device_stats.clear()
+        self.device_time_series.clear()
+
+    def _get_bids_offers_trades(self, area_uuid: str, market_type: int,
+                                market_stats: Dict[str, Dict]):
+        for time_slot, orders in market_stats.items():
+            time_slot_dt = from_format(time_slot, DATE_TIME_FORMAT)
+            self.orders[area_uuid][market_type][time_slot_dt] = {
+                order: orders.get(order, []) for order in ("offers", "bids", "trades")}
+
+    def _update_stats_and_time_series(self, forward_results: Dict, market_type_value: int):
+        market_type = AvailableMarketTypes(market_type_value)
+        for time_slot, device_results in forward_results.items():
+            for device_uuid, current_device_stats in device_results.items():
+                if previous_forward_stats := self.previous_device_stats.get(
+                        market_type_value, {}).get(time_slot, {}).get(device_uuid, {}):
+                    previous_device_stats = ForwardDeviceStats.from_dict(previous_forward_stats)
+                    current_device_stats += previous_device_stats
+                self.current_device_stats[market_type_value][time_slot][device_uuid] = \
+                    current_device_stats.to_dict()
+                if market_type not in ALLOWED_MARKET_TYPES:
+                    # TODO: delete this check, when ForwardTradeProfileGenerator profile generation
+                    #  functionality will be extend with all forward markets types
+                    continue
+                for resolution in MARKET_RESOLUTIONS.get(market_type, []):
+                    device_time_series = ForwardDeviceTimeSeries(current_device_stats, market_type)
+                    all_time_series_generators = device_time_series.generate(
+                        resolution=resolution.duration())
+                    all_time_series = {k: dict(v) for k, v in all_time_series_generators.items()}
+                    self.device_time_series[market_type_value][resolution.value][time_slot][
+                        device_uuid] = all_time_series
```

Comment:
IMO, for better readability, this part could be refactored into a separate function. What do you hink?